mm2src/mm2_metrics/src/recorder.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{native::MetricNameValueMap, MetricType, MetricsJson};
+
+use common::log::{error, Tag};
+use fomat_macros::wite;
+use metrics::{Counter, Gauge, Key, KeyName, Label, Recorder, Unit};
+use metrics_util::registry::{GenerationalAtomicStorage, Registry};
+use std::fmt::Write;
+use std::{collections::HashMap, slice::Iter, sync::atomic::Ordering};
+
+pub struct MmRecorder {
+    pub(crate) registry: Registry<Key, GenerationalAtomicStorage>,
+}
+
+impl Default for MmRecorder {
+    fn default() -> Self {
+        Self {
+            registry: Registry::new(metrics_util::registry::GenerationalStorage::atomic()),
+        }
+    }
+}
+
+impl MmRecorder {
+    pub fn prepare_json(&self) -> MetricsJson {
+        let mut output = vec![];
+
+        for counters in self.registry.get_counter_handles() {
+            let (key, labels) = counters.0.into_parts();
+            let value = counters.1.get_inner().load(Ordering::Acquire);
+            output.push(MetricType::Counter {
+                key: key.as_str().to_string(),
+                labels: labels_into_parts(labels.clone().iter()),
+                value,
+            });
+        }
+
+        for counters in self.registry.get_gauge_handles() {
+            let (key, labels) = counters.0.into_parts();
+            let value = f64::from_bits(counters.1.get_inner().load(Ordering::Acquire));
+            output.push(MetricType::Gauge {
+                key: key.as_str().to_string(),
+                labels: labels_into_parts(labels.clone().iter()),
+                value,
+            });
+        }
+
+        for (key, value) in self.registry.get_histogram_handles() {
+            let value = value.get_inner().data().iter().sum::<f64>();
+            let (key, labels) = key.into_parts();
+            let mut qauntiles_value = HashMap::new();
+            qauntiles_value.insert(key.as_str().to_string(), value as u64);
```

Comment:
This is not right. `MetricType::Histogram::quantiles` is declared as a `HashMap` container for some reason.
Please check what's the reason and how this part (converting a histogram to `MetricType`) was implemented previously